source/lac/petsc_vector_base.cc

```diff
@@ -724,6 +793,9 @@ namespace PETScWrappers
     const PetscErrorCode ierr = VecScale(vector, a);
     AssertThrow(ierr == 0, ExcPETScError(ierr));
 
+    // update ghost values if needed
+    update_ghost_values();
```

Comment:
This is in function `operator *= (const PetscScalar a)`, which is a function that changes values of the vector and as such we treat it as a function that cannot have ghost entries. Indeed, the function starts with
```
Assert(!has_ghost_elements(), ExcGhostsPresent());
```
As a consequence, there is no need to update ghost values -- or am I missing something fundamental?

This is the case for several of the other functions below as well.

Reply:
I left it like this in case we decide we want to make ghosted vectors not read-only

Reply:
@bangerth @luca-heltai can we merge this? In the next days, I will open a PR for `SNES`

Reply:
I don't think you've addressed the question I had. Basically, it comes down to this: The function reads
```VectorBase &
  VectorBase::operator*=(const PetscScalar a)
  {
    Assert(!has_ghost_elements(), ExcGhostsPresent());
    AssertIsFinite(a);
    const PetscErrorCode ierr = VecScale(vector, a);
    AssertThrow(ierr == 0, ExcPETScError(ierr));

    // update ghost values if needed
    update_ghost_values();

   return *this;
 }
```
So:
* At the very top, we assert that the function has no ghost entries.
* We then do updates on the vector elements, which by the previous point can only be locally-owned elements.
* Then is the `update_ghost_values()` call, but that shouldn't do anything because *we already know that there are no ghost values*.

I just don't understand why the `update_ghost_values()` call should be there. I find that confusing, and I think the line should be removed here and in all of the other functions that assert that there are no ghost entries.

Reply:
My idea was to leave the call there (and in all other places where it was needed) in case in the future you will allow ghosted values to not be read-only.  I can remove it (or comment it out) if you don't like it or find it confusing. Tell me what you prefer and I'll do it

Reply:
Let's remove it. It currently serves no purpose, and it can only confuse readers of the code (like it did confuse me).

Reply:
I agree. If you want to support modifiying ghosted vectors in the future, you don't want to update the ghost values every time you do some operation. This would cause unnecessary work when say running an iterative solver.

This is by the way why we introduced the notion of read-only ghosted vectors: You either potentially have a lot of additional overhead if you accidentally use a ghosted vector inside a linear solver or (if you only update the ghost values in a separate call to ``update_ghost_values``) you have the potential for bugs because ghost values can be out of date.

Reply:
done

Reply:
@bangerth @tjhei I have removed the calls